vignettes/sparsegl.Rmd

```diff
@@ -0,0 +1,332 @@
+---
+title: Get started with `sparsegl`
+description: An introductory tutorial with examples
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Get started with `sparsegl`}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This package provides tools for fitting the regularization paths for sparse group-lasso
+penalized learning problems. The model is fit for a sequence of the regularization
+parameters lambda.
+
+The strengths and improvements that this package achieved compared to other sparse group-lasso
+packages are as follows: 
+
+* `X` can be a sparse input matrix passed as an argument in the main function 
+`sparsegl()`, and it does not have to be a regular matrix.
+
+* plots are made with package `ggplot2` instead of function plot() from `base`, 
+which provide better appearances and experience for visualization.
+
+* An extra function `risk-estimate()` evaluates the quality of fitted models in
+terms of different information criteria and provides a means for model selection.
+
+
+## Installing
+
+This package is not on CRAN yet, so it can be installed using the [`devtools`] 
+(https://cran.r-project.org/package=devtools) package:
+
+```{r, eval = FALSE}
+devtools::install_github("dajmcdon/sparsegl", ref = "main")
+```
+
+Building the vignettes, such as this getting started guide, takes a significant
+amount of time. They are not included in the package by default. If you want to
+include vignettes, then use this modified command:
+
+```{r, eval = FALSE}
+devtools::install_github("dajmcdon/sparsegl", ref = "main",
+                          build_vignettes = TRUE, dependencies = TRUE)
+```
+
+For this getting-started vignette, firstly, we will randomly generate X, an input matrix of 
+predictors of dimension n-obs by p-feature. To initiate y, namely a vector of real-value 
+responses of length n, or a matrix with 1 colnmn with
+$$
+y = X\beta^* + \epsilon,
+$$
+where the coefficient vector $\beta^*$ is specified as below, and the white noise
+$\epsilon$ following standard normal distribution serves as data variation.
+
+```{r}
+library(sparsegl)
+set.seed(1010)
+n <- 100
+p <- 200
+X <- matrix(data = rnorm(n*p, mean = 0, sd = 1), nrow = n, ncol = p)
+eps <- rnorm(n, mean = 0, sd = 1)
+beta_star <- c(rep(5, 5), c(5, -5, 2, 0, 0), rep(-5, 5), c(2, -3, 8, 0, 0), rep(0, (p - 20)))
+y <- X %*% beta_star + eps
+groups <- rep(1:(p / 5), each = 5)
+```
+
+## Function `sparsegl()`
+
+`sparsegl()` is the main function in the `sparsegl` package. Given an input matrix 
+X, and a response vector y (or a matrix with 1 column), a sparse group-lasso 
+regularized linear model is fit for a sequence of penalty parameter values in 
+terms of penalized maximum likelihood. Other arguments of this function can be 
+optionally initialized differently than the default values, and the main arguments are:
```

Comment:
Here and below, there's no need to state what every argument does. Just the novel ones. For `sparsegl()`, you should explain `asparse`.

Reply:
Then show the output, and say what `df` is.